### PR TITLE
Bump torch to 2.1.1 version

### PR DESCRIPTION
**Pull request by j316chuck**
# What does this PR do?
Bump base torch to 2.1.1 

# What issue(s) does this change relate to?
https://databricks.atlassian.net/browse/UWU-2134

### CODE REVIEW

**Review by mvpatel2000 — COMMENTED** [review]
You need to also bump setup.py and the mosaic_fsdp wrap. I think we can just remove the `else` clause since we need no more monkeypatches CC: @b-chu

**Comment by j316chuck** [comment]
Bumped up setup.py. Let's not remove else since we might need a docker image down the line for @eracah's DTensor work.

**Comment by j316chuck** [comment]
Oh i thought the else was for nightly else.

File: composer/trainer/mosaic_fsdp.py [hunk 1]
```diff
@@ -50,5 +50,6 @@ def patch_pytorch():
         # Monkey patch partial state dict handling
         _state_dict_utils._sharded_pre_load_state_dict_hook = (_sharded_pre_load_state_dict_hook)
 
-    elif version.parse(torch.__version__) >= version.parse('2.1.1'):
-        raise NotImplementedError(f'FullyShardedDataParallel is not supported for torch >= 2.2.0')
+    else:
+        # Nothing to patch yet for torch >= 2.1.1
+        pass
```
Comment (mvpatel2000) [inline composer/trainer/mosaic_fsdp.py:55]:
```suggestion
```

**Review by mvpatel2000 — APPROVED** [review]
LGTM. Can we remove `else pass` though

**Pull request merged**